### PR TITLE
Added !leaderboard

### PR DESCRIPTION
**Pull request by byn3**
Usage:  "!leaderboard <positive int>"
bad inputs gives error messages. formatted and lined up the final printed string 

bad inputs I thought of:
NaN
<= 0
requesting more than the roster size

### CODE REVIEW

File: commands/leaderboard.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+module.exports = {
+  leaderboard: function (message, roster) {
+
+    if(message.content.split(" ").length !== 2) { // we want users to input how many
+      message.channel.send("Please use like so: !leaderboard <positive integer>");
+      return;
+    }
+
+    let stringy = "List of beautiful whales <3\n```";
```
Comment (tommymurphy) [inline commands/leaderboard.js]: Remove this :P

File: commands/leaderboard.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    let amount = message.content.split(" ")[1];
+    amount = Math.min(amount, roster.length); //prevents out of index errors
+
+    if (!parseInt(amount)) { //for NaN inputs
+      message.channel.send("Use a positive WHOLE number you twat");
```
Comment (tommymurphy) [inline commands/leaderboard.js]: Make this message more nice and fit the Zakumbot language

File: commands/leaderboard.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      return;
+    }
+
+    if (parseInt(amount) < 1) { // for 0 or negatives
+      message.channel.send("Stop it. Use a positive WHOLE number");
```
Comment (tommymurphy) [inline commands/leaderboard.js]: This too

File: commands/leaderboard.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      return;
+    }
+
```
Comment (tommymurphy) [inline commands/leaderboard.js]: Shorter JS/ES6 version of lines 25-38:
```suggestion
message.channel.send(`\`\`\`${roster.sort((a, b) => b.rank - a.rank)
      .map((member,index) => `#${index+1} ${member.name} ${" ".repeat(Math.abs(member.name.length - 13 + (index+1).toString().length))} ${member.rank}`)
      .slice(0,amount)
      .join('\n')}\`\`\``)
```

Comment (tommymurphy) [reply commands/leaderboard.js]: You might have to add a .repeat after the ${index+1} though to make the spacing more consistent again

File: commands/leaderboard.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+var fs = require("fs");
```
Comment (tommymurphy) [inline commands/leaderboard.js]: Not used

File: commands/leaderboard.js (hunk 2 above)
Comment (tommymurphy) [reply commands/leaderboard.js]: You can combine this if with the other and just use a generic and nicer error

File: commands/leaderboard.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+module.exports = {
+  leaderboard: function (message, roster) {
+
+    if(message.content.split(" ").length !== 2) { // we want users to input how many
```
Comment (tommymurphy) [inline commands/leaderboard.js]: Let's make the default number `10` - add the command spec to commands.json to define how to use the command

Comment (byn3) [reply commands/leaderboard.js]: done.
thank you for doing the needful.

File: commands/leaderboard.js (hunk 4 above)
Comment (byn3) [reply commands/leaderboard.js]: this is perfect. modified a bit and it is all evenly spaced and nice.
thank u kind sur. much smart. many brain.

**Comment by byn3** [comment]
new PR up. added all your requested changes. it looks good

**Pull request merged**